var/spack/repos/builtin/packages/zfp/package.py

```diff
@@ -62,53 +73,82 @@ class Zfp(CMakePackage, CudaPackage):
+        "all zeroes",
     )
-    variant("aligned", default=False, description="Enable aligned memory allocation")
-    variant("twoway", default=False, description="Use two-way skew-associative cache")
+
     variant(
         "fasthash",
         default=False,
         description="Use a faster but more collision prone hash function",
     )
+
     variant("profile", default=False, description="Count cache misses")
 
+    variant(
+        "round",
+        default="ZFP_ROUND_NEVER",
+        values=("ZFP_ROUND_NEVER", "ZFP_ROUND_FIRST", "ZFP_ROUND_LAST"),
+        multi=False,
+        description="EXPERIMENTAL: Set coefficient rounding method",
+    )
+
+    variant(
+        "strided", default=False, description="Enable strided access for progressive zfp streams"
+    )
+
+    variant(
+        "tight_error",
+        default=False,
+        description="EXPERIMENTAL: Enable rounding-based error reduction",
+    )
+
+    variant("twoway", default=False, description="Use two-way skew-associative cache")
+
     # Conflicts
     conflicts("+c", when="@:0.5.3", msg="+c requires zfp 0.5.4 or later")
-    conflicts("+python", when="@:0.5.4", msg="+python requires zfp 0.5.5 or later")
-    conflicts("+fortran", when="@:0.5.4", msg="+fortran requires zfp 0.5.5 or later")
-    conflicts("+openmp", when="@:0.5.2", msg="+openmp requires zfp 0.5.3 or later")
-    conflicts("+cuda", when="@:0.5.3", msg="+cuda requires zfp 0.5.4 or later")
+    conflicts("+daz", when="@:0.5.5", msg="+daz required zfp 1.0.0 or later")
+    conflicts("+python", when="@:0.5.4", msg="+c requires zfp 0.5.4 or later")
+    conflicts("+fortran", when="@:0.5.4", msg="+c requires zfp 0.5.4 or later")
+    conflicts("+openmp", when="@:0.5.2", msg="+c requires zfp 0.5.4 or later")
+    conflicts("+cuda", when="@:0.5.3", msg="+c requires zfp 0.5.4 or later")
     conflicts("+fasthash", when="@:0.5.1", msg="+fasthash requires zfp 0.5.2 or later")
     conflicts("+profile", when="@:0.5.1", msg="+profile requires zfp 0.5.2 or later")
+    conflicts("round", when="@:0.5.5", msg="+round requires zfp 1.0.0 or later")
+    conflicts("+tight_error", when="@:0.5.5", msg="+tight_error requires zfp 1.0.0 or later")
+    conflicts(
+        "+tight_error",
+        when="round=ZFP_ROUND_NEVER",
+        msg="Using zfp with tight error requires a rounding mode other than ZFP_ROUND_NEVER",
+    )
 
-    # Dependencies
-    depends_on("cmake@3.4.0:", type="build")
-    depends_on("cuda@7:", type=("build", "test", "run"), when="+cuda")
-    depends_on("python", type=("build", "test", "run"), when="+python")
-    depends_on("py-numpy", type=("build", "test", "run"), when="+python")
-    depends_on("py-cython", type="build", when="+python")
-
+    # CMake options
     def cmake_args(self):
         spec = self.spec
 
-        # CMake options
         args = [
-            # TODO: self.define_from_variant('BUILD_UTILITIES', 'utilities'),
             self.define("BUILD_TESTING", self.run_tests),
```

Comment:
Out of curiosity, how will this interact with the new `BUILD_TESTING_FULL` on `develop`?

Reply:
This should act identically. Its just that prior to 1.0.0 we stripped tests out so what Spack was running is the same set of tests that BUILD_TESTING runs now. Develop would have run whats in BUILD_TESTING_FULL but that wasn't as far as I'm aware done for any reason. It happened automatically because of how tests were managed. We could add BUILD_TESTING_FULL as an option but as its intended for zfp development work I'd question the need.